examples/linear_deformation.py

```diff
@@ -0,0 +1,1187 @@
+# Copyright 2014-2016 The ODL development group
```

Comment:
We need to find a place where to put all the deformation related code. I suggest that we make a new subpackage `deform` under `odl/`.